src/detect-byte-extract.c

```diff
@@ -391,9 +392,10 @@ SigMatch *DetectByteExtractRetrieveSMVar(const char *arg, int sm_list, const Sig
         SigMatch *sm = s->init_data->smlists[list];
         while (sm != NULL) {
             // Make sure that the linked buffers ore on the same list
-            if (sm->type == DETECT_BYTE_EXTRACT && (sm_list == -1 || sm_list == list)) {
+            if (sm->type == DETECT_BYTE_EXTRACT) {
                 const SCDetectByteExtractData *bed = (const SCDetectByteExtractData *)sm->ctx;
                 if (strcmp(bed->name, arg) == 0) {
+                    *found_list = list;
```

Comment:
This change looks suspicious

Did you test a rule with 2 buffers, each buffer doing a byte_extract+ byte_math ?

I am afraid that with this change, the second buffer will get here `found_list` from the first buffer when it uses to get the right  `sm_list == list` in current code